server/index.js

```diff
@@ -58,6 +58,126 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json({});
+  } else {
+    const sql = `
+    select "c"."cartItemId",
+      "c"."price",
+      "p"."productId",
+      "p"."image",
+      "p"."name",
+      "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using("productId")
+      where "c"."cartId" = $1
+    `;
+
+    const params = [req.session.cartId];
+
+    db.query(sql, params)
+      .then(result => {
+        const cart = result.rows;
+        res.status(200).json({ cart });
+      });
+  }
+
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId, 10);
+  const cartOfPerson = req.session.cartId;
+  if (productId < 0) {
+    res.status(400).json({
+      error: 'productId must be a postive integer'
+    });
+  }
+
+  const sql = `
+    select "price"
+    from "products"
+    where "productId" = $1
+  `;
+
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const productPrice = result.rows[0];
+      if (!productPrice) {
+        throw res.status(400).json({
+          error: 'No results found'
+        });
+      }
+      if (!req.session.cartId) {
+        const sql = `
+        insert into "carts" ("cartId","createdAt")
+        values(default, default)
+        returning "cartId"
+          `;
+        return db.query(sql)
+          .then(result => result.rows)
+          .then(result => {
+            return {
+              productPrice,
+              result
+            };
+          });
+      }
+      return {
+        productPrice,
+        cartOfPerson
+      };
+    })
+    .then(result => {
+      if (cartOfPerson !== req.session.cartId) {
+        req.session.cartId = result.result[0].cartId;
+      }
+      const price = result.productPrice.price;
+      const sql = `
+      insert into "cartItems" ("cartId","productId", "price")
+      values ($1, $2, $3)
+      returning "cartItemId"
+      `;
+
+      const params = [cartOfPerson, productId, price];
+      return db.query(sql, params)
+        .then(result => {
+          const cart = result;
+          return cart;
+        });
+    })
+    .then(result => {
+
+      const sql = `
+        select "c"."cartItemId",
+        "c"."price",
+        "p"."productId",
+        "p"."image",
+        "p"."name",
+        "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+        where "c"."cartItemId" = $1
+        `;
+
+      const params = [result.rows[0].cartItemId];
+
+      db.query(sql, params)
```

Comment:
Be sure to also `return` this `db.query()` so that if something goes wrong your final `.catch()` can handle the error